include/Dialect/BGV/IR/BGVOps.td

```diff
@@ -134,7 +140,7 @@ def BGV_ExtractOp : BGV_Op<"extract", [AllTypesMatch<["input", "output"]>]> {
     RLWECiphertext:$output
   );
 
-  let assemblyFormat = "operands attr-dict `:` qualified(type($input)) `,` type($offset)" ;
+  let hasVerifier = 1;
```

Comment:
just FYI for later, if you use the TypesMatchWith style thing, you get the inferred builder and you could have a shortened syntax in the same way things like memref.load have (so you only print the input shaped type and the output is inferred)

Reply:
I don't think that trait is enough to infer types, it's just a CPred. I think what appears to be type inference in memref.load is because the output is an AnyType and the verifier handles the rest.